validphys2/src/validphys/photon/compute.py

```diff
@@ -0,0 +1,312 @@
+
+    def set_thresholds_alpha_em(self):
+        """
+        Compute and store the couplings at thresholds to speed up the calling
+        to alpha_em inside fiatlux:
+        when q is in a certain range (e.g. thresh_c < q < thresh_b) and qref in a different one
+        (e.g. thresh_b < q < thresh_t) we need to evolve from qref to thresh_b with nf=5 and then
+        from thresh_b to q with nf=4. Given that the value of alpha at thresh_b is always the same
+        we can avoid computing the first step storing the values of alpha in the threshold points.
+        It is done for qref in a generic range (not necessarly qref=91.2).
+
+        """
+        self.thresh_c = self.theory["kcThr"] * self.theory["mc"]
+        self.thresh_b = self.theory["kbThr"] * self.theory["mb"]
+        self.thresh_t = self.theory["ktThr"] * self.theory["mt"]
+        if self.theory["MaxNfAs"] <= 5:
+            self.thresh_t = np.inf
+        if self.theory["MaxNfAs"] <= 4:
+            self.thresh_b = np.inf
+        if self.theory["MaxNfAs"] <= 3:
+            self.thresh_c = np.inf
+
+        thresh_list = [self.thresh_c, self.thresh_b, self.thresh_t]
+        # determine nfref
+        if self.qref < self.thresh_c:
+            nfref = 3
+        elif self.qref < self.thresh_b:
+            nfref = 4
+        elif self.qref < self.thresh_t:
+            nfref = 5
+        else:
+            nfref = 6
+        thresh_list.insert(nfref - 3, self.qref)
+
+        self.thresh = {nf: thresh_list[nf - 3] for nf in range(3, self.theory["MaxNfAs"] + 1)}
+
+        self.alpha_thresh = {nfref: self.alpha_em_ref}
+
+        # determine the values of alpha in the threshold points, depending on the value of qref
+        for nf in range(nfref + 1, self.theory["MaxNfAs"] + 1):
+            self.alpha_thresh[nf] = self.alpha_em_nlo(
+                self.thresh[nf], self.alpha_thresh[nf - 1], self.thresh[nf - 1], nf - 1
+            )
+
+        for nf in reversed(range(3, nfref)):
+            self.alpha_thresh[nf] = self.alpha_em_nlo(
+                self.thresh[nf], self.alpha_thresh[nf + 1], self.thresh[nf + 1], nf + 1
+            )
+
+    def set_betas(self):
+        """Compute and store beta0 / 4pi and b1 = (beta1/beta0)/4pi as a function of nf."""
+        nl = 3
+        nc = 3
+        eu2 = 4.0 / 9
+        ed2 = 1.0 / 9
+        self.beta0 = {}
+        self.b1 = {}
+        for nf in range(3, 6 + 1):
+            nu = nf // 2
+            nd = nf - nu
+            self.beta0[nf] = (-4.0 / 3 * (nl + nc * (nu * eu2 + nd * ed2))) / (4 * np.pi)
+            self.b1[nf] = (
+                -4.0
+                * (nl + nc * (nu * eu2**2 + nd * ed2**2))
+                / self.beta0[nf]
+                / (4 * np.pi) ** 2
+            )
+
+    def compute_photon_array(self, id):
+        r"""
+        Compute the photon PDF for every point in the grid xgrid.
+
+        Parameters
+        ----------
+        id: int
+            replica id
```

Comment:
`id`, `replica_id` and `replica` are all used to indicate the same thing. I think in most of the n3fit code simply `replica` it makes things more readable if it's called the same everywhere.